openedx/core/djangoapps/discussions/api/providers.py

```diff
@@ -0,0 +1,29 @@
+from typing import List, Optional
+
+from opaque_keys.edx.keys import LearningContextKey
+
+from .config import get_discussion_config
+from ..discussions_apps import DiscussionApp, DiscussionAppsPluginManager
+
+
+def get_discussion_provider(context_key: LearningContextKey) -> Optional[DiscussionApp]:
+    """
+    Returns the discussion app provider associated with the provided context key.
+
+    Args:
+        context_key (LearningContextKey): Learning context, currently only a course
+
+    Returns:
+        A DiscussionApp instance for the discussion provider associated with this context.
+    """
+
+    config = get_discussion_config(context_key)
+    if config:
+        return DiscussionAppsPluginManager.get_plugin(config.provider)
```

Comment:
This does not seem to be working for me. Entering http://localhost:18010/tabs/course-v1:edX+DemoX+Demo_Course returns:
```python

Traceback:

File "/edx/app/edxapp/venvs/edxapp/lib/python3.5/site-packages/django/core/handlers/exception.py" in inner
  34.             response = get_response(request)

File "/edx/app/edxapp/venvs/edxapp/lib/python3.5/site-packages/django/core/handlers/base.py" in _get_response
  115.                 response = self.process_exception_by_middleware(e, request)

File "/edx/app/edxapp/venvs/edxapp/lib/python3.5/site-packages/django/core/handlers/base.py" in _get_response
  113.                 response = wrapped_callback(request, *callback_args, **callback_kwargs)

File "/usr/lib/python3.5/contextlib.py" in inner
  30.                 return func(*args, **kwds)

File "/edx/app/edxapp/edx-platform/common/djangoapps/util/json_request.py" in parse_json_into_request
  55.         return view_function(request, *args, **kwargs)

File "/edx/app/edxapp/venvs/edxapp/lib/python3.5/site-packages/django/contrib/auth/decorators.py" in _wrapped_view
  21.                 return view_func(request, *args, **kwargs)

File "/edx/app/edxapp/venvs/edxapp/lib/python3.5/site-packages/django/utils/decorators.py" in _wrapped_view
  142.                     response = view_func(request, *args, **kwargs)

File "/edx/app/edxapp/venvs/edxapp/lib/python3.5/site-packages/django/views/decorators/http.py" in inner
  40.             return func(request, *args, **kwargs)

File "/edx/app/edxapp/edx-platform/cms/djangoapps/contentstore/views/tabs.py" in tabs_handler
  67.                                                      include_hidden=True):

File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/tabs.py" in iterate_displayable
  466.             if tab.is_enabled(course, user=user) and (include_hidden or not (user and tab.is_hidden)):

File "/edx/app/edxapp/edx-platform/openedx/core/djangoapps/discussions/tabs.py" in is_enabled
  62.         provider = get_discussion_provider(course.id)

File "/edx/app/edxapp/edx-platform/openedx/core/djangoapps/discussions/api/providers.py" in get_discussion_provider
  22.         return DiscussionAppsPluginManager.get_plugin(config.provider)

File "/edx/app/edxapp/edx-platform/openedx/core/lib/plugins.py" in get_plugin
  45.                 namespace=namespace or cls.NAMESPACE,  # pylint: disable=no-member

Exception Type: PluginError at /tabs/course-v1:edX+DemoX+Demo_Course
Exception Value: No such plugin cs_comments for entry point openedx.discussion_apps
```

After enabling `Test app` via Django admin it gives me:
```python
Traceback (most recent call last):
  File "/edx/app/edxapp/edx-platform/cms/djangoapps/contentstore/views/preview.py", line 327, in get_preview_fragment
    fragment = module.render(preview_view, context)
  File "/edx/app/edxapp/venvs/edxapp/lib/python3.5/site-packages/xblock/core.py", line 202, in render
    return self.runtime.render(self, view, context)
  File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/x_module.py", line 1979, in render
    return self.__getattr__('render')(block, view_name, context)
  File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/x_module.py", line 1423, in render
    return super(MetricsMixin, self).render(block, view_name, context=context)
  File "/edx/app/edxapp/venvs/edxapp/lib/python3.5/site-packages/xblock/runtime.py", line 829, in render
    frag = view_fn(context)
  File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/html_module.py", line 93, in student_view
    fragment = Fragment(self.get_html())
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/edxnotes/decorators.py", line 41, in get_html
    user = self.runtime.get_real_user(self.runtime.anonymous_student_id)
TypeError: 'NoneType' object is not callable
```

Reply:
You need to run `pip install -e .` in your edx-platform directory after doing this since otherwise the plugins won't be registered.